editor/src/components/canvas/canvas-strategies/flow-reorder-helpers.ts

```diff
@@ -33,6 +29,55 @@ export function isValidFlowReorderTarget(
   }
 }
 
+export function areAllSiblingsInOneDimension(
+  target: ElementPath,
+  metadata: ElementInstanceMetadataMap,
+): boolean {
+  const targetElement = MetadataUtils.findElementByElementPath(metadata, target)
+  if (targetElement == null) {
+    return false
+  }
+
+  const siblings = MetadataUtils.getSiblings(metadata, target)
+  const targetDirection = getElementDirection(targetElement)
+
+  let allHorizontalOrVertical = true
+  let frames: Array<CanvasRectangle> = []
+  fastForEach(siblings, (sibling) => {
+    if (isValidFlowReorderTarget(sibling.elementPath, metadata)) {
+      if (getElementDirection(sibling) !== targetDirection) {
+        allHorizontalOrVertical = false
+      }
+      if (sibling.globalFrame != null) {
+        frames.push(sibling.globalFrame)
+      }
+    }
+  })
+
+  return allHorizontalOrVertical && areNonWrappingSiblings(frames, targetDirection)
+}
+
+function areNonWrappingSiblings(
+  frames: Array<CanvasRectangle>,
+  direction: 'horizontal' | 'vertical',
+): boolean {
+  // wrapping or columns set: a sibling frame has a larger x/y then the right/bottom edge of its siblings
+  return frames.every((frame) => {
+    return frames.every((frameToCompare) => {
+      if (direction === 'horizontal') {
+        return frameToCompare.y < frame.y + frame.height
```

Comment:
I'm not sure but isn't horizontal and vertical swapped here? shouldn't horizontal be comparing the x coordinate?

Reply:
the horizontal direction is for the layout type, I fixed the variable name here. What it checks here is if the elements are in an inline layout with wrapping texts some siblings will have their y below the siblings bottom edge

Reply:
> hmmmm I'm not sure I like that check, because it would break for this case: <img alt="image" width="341" src="https://user-images.githubusercontent.com/2226774/193594510-40393aeb-03e5-45b2-a3c1-abb00d1bb276.png">
> 
> I think it's more reliable and less strict to verify that in a left-to-right row layout, no element should be to the left to their preceding sibling. so if element n wraps, that will jump to the left edge of the layout, meaning it will be to the left of element n-1. <img alt="image" width="737" src="https://user-images.githubusercontent.com/2226774/193595795-60fb7465-c19e-4969-8c1f-b231e9b721ad.png">

the second version is nicer and less strict, but for the 1st image how do you know where to drag if we use the starting frames?

Reply:
> the second version is nicer and less strict, but for the 1st image how do you know where to drag if we use the starting frames?

if we know that we are thinking in terms of a 1-dimensional restricted drag, we could compare just the X or just the Y coordinates. so in this case, the entire area divided by red could trigger the reorder, not just the element itself:
<img width="1092" alt="image" src="https://user-images.githubusercontent.com/2226774/193597623-93e60e7e-bdfc-42b5-95c6-26ed0db2ddd4.png">


or a bit more strictly like this:
<img width="1101" alt="image" src="https://user-images.githubusercontent.com/2226774/193597684-a5c2a2f5-4e6e-43ec-b3ab-901a4a2ed8b1.png">